packages/dev/docs/src/client.js

```diff
@@ -183,8 +185,95 @@ function Hamburger() {
   );
 }
 
+function DocSearch() {
+  const client = algoliasearch('1V1Q59JVTR', '44a7e2e7508ff185f25ac64c0a675f98');
+  const searchIndex = client.initIndex('react-spectrum');
+  const searchOptions = {
+    highlightPreTag: `<mark class="${docsStyle.docSearchBoxMark}">`,
+    highlightPostTag: '</mark>'
+  };
+
+  const [searchValue, setSearchValue] = useState('');
+  const [predictions, setPredictions] = useState(null);
+  const [suggestions, setSuggestions] = useState(null);
+
+  let updatePredictions = ({hits}) => {
+    setPredictions(hits);
+    let sections = [];
+    hits.forEach(prediction => {
+      let hierarchy = prediction.hierarchy;
+      let objectID = prediction.objectID;
+      let lvl0 = hierarchy.lvl0;
+      let section = sections.find(section => section.title === lvl0);
+      if (!section) {
+        section = {title: lvl0, items: []};
+        sections.push(section);
+      }
+      let text = [];
+      let textValue = [];
+      for (let i = 1; i < 6; i++) {
+        if (hierarchy[`lvl${i}`]) {
+          text.push(prediction._highlightResult.hierarchy[`lvl${i}`].value);
+          textValue.push(hierarchy[`lvl${i}`]);
+        }
+      }
+      section.items.push(
+        <Item key={objectID} textValue={textValue.join(' | ')}>
+          <Text><span dangerouslySetInnerHTML={{__html: text.join(' | ')}} /></Text>
+          {
+            prediction.content &&
+            <Text slot="description">
+              <span dangerouslySetInnerHTML={{__html: prediction._snippetResult.content.value}} />
+            </Text>
+          }
+        </Item>
+      );
+    });
+    let suggestions = sections.map((section, index) => <Section key={`${index}-lvl0-${section.title}`} title={section.title}>{section.items}</Section>);
+    setSuggestions(suggestions);
+  };
+
+  let onInputChange = (query) => {
+    if (!query && predictions) {
+      setPredictions(null);
+      setSuggestions(null);
+    }
+    setSearchValue(query);
+    searchIndex
+      .search(
+        query,
+        searchOptions
+      )
+      .then(updatePredictions);
+  };
+
+  let onSubmit = (value, key) => {
+    if (key) {
+      let prediction = predictions.find(prediction => key === prediction.objectID);
+      let url = prediction.url;
+      window.location.href = `${window.location.hostname === 'reactspectrum.blob.core.windows.net' ? window.location.href.replace(/(.+\/docs\/)(.+)/, '$1') : '/'}${url.replace('https://react-spectrum.adobe.com/', '')}`;
+    }
+  };
+
+  return (
+    <Provider theme={defaultTheme} role="search">
```

Comment:
why do we need this provider? I don't see one for Hamburger

Reply:
`SearchAutocomplete` is a React-Spectrum component it throws an error when `useProvider` props like `scale` are unavailable.

Reply:
but Hamburger uses ActionButton? so that should have the same issue right?

Reply:
`ActionButton` does not [call `useListBoxLayout`](https://github.com/adobe/react-spectrum/blob/425afd51e8e2117be37d83cdca15bbe8d869d60d/packages/%40react-spectrum/autocomplete/src/SearchAutocomplete.tsx#L89), which in turn [requires a `Provider`](https://github.com/adobe/react-spectrum/blob/425afd51e8e2117be37d83cdca15bbe8d869d60d/packages/%40react-spectrum/listbox/src/ListBoxBase.tsx#L51).

Reply:
Ah, ok. I should use `ThemeProvider` from `ThemeSwitcher.js` instead.